File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/service/server/MeteoriteServerException.java
package com.antonio.samir.meteoritelandingsspots.service.server;

/**
 * Server Exception created in order to prevent the ui layer be aware the exception of service layer
 */
public class MeteoriteServerException extends Exception {

    public MeteoriteServerException(final Exception exception) {
        super(exception);
    }

    public MeteoriteServerException(final String exception) {
        super(exception);
    }

}

File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/data/remote/MeteoriteServerException.kt
package com.antonio.samir.meteoritelandingsspots.data.remote

import com.antonio.samir.meteoritelandingsspots.common.DataError

/**
 * Wraps transport failures so callers above the data layer never see Retrofit or OkHttp types.
 */
class MeteoriteServerException(
    val error: DataError,
    cause: Throwable? = null,
) : Exception("Meteorite feed unavailable: $error", cause)
